src/wirecraft/client/game.py

```diff
@@ -240,41 +262,30 @@ def update_zoom(self):
     def handle_left_click(self, camera: Camera) -> None:
         """Handle left mouse button click."""
         if not self.is_placing_cable:
-            pass
-            # self.start_cable_connection(camera)
+            self.start_cable_connection(camera)
         else:
-            pass
-            # self.end_cable_connection(camera)
+            self.end_cable_connection(camera)
         self.dragging = True
         self.last_mouse_pos = pygame.mouse.get_pos()
         for device in self.devices:
             device.update_position(camera, self.resolution.size)
 
-    # def start_cable_connection(self, camera: Camera) -> None:
-    #     """Start a cable connection."""
-    #     for device in self.devices:
-    #         if device.get_rect().collidepoint(pygame.mouse.get_pos()):
-    #             self.cables.append(
-    #                 Cable(
-    #                     camera.world_to_screen(device.centered_world_position, self.resolution.size),
-    #                     pygame.mouse.get_pos(),
-    #                 )
-    #             )
-    #             self.is_placing_cable = True
-    #             break
-
-    # def end_cable_connection(self, camera: Camera) -> None:
-    #     """End a cable connection."""
-    #     for device in self.devices:
-    #         if (
-    #             device.get_rect().collidepoint(pygame.mouse.get_pos())
-    #             and camera.world_to_screen(device.centered_world_position, self.resolution.size)
-    #             != self.cables[-1].start
-    #         ):
-    #             self.is_placing_cable = False
-    #             self.cables[-1].ended = True
-    #             self.cables[-1].end = camera.world_to_screen(device.centered_world_position, self.resolution.size)
-    #             break
+    # TODO: do not hardcode port 0
+    def start_cable_connection(self, camera: Camera) -> None:
+        """Start a cable connection."""
+        for device in self.devices:
+            if device.screen_rect.collidepoint(pygame.mouse.get_pos()):
+                self.is_placing_cable = True
+                self.server.add_cable(device.db_id, 0, -1, -1, LEVEL)
```

Comment:
Hmm, we could check from the server if the port is available, but maybe we don't need to create the cable while it is not wired from point A to B ?
but maybe it's ok.

Reply:
Should definitely check if the port is occupied, good catch. For the creation of cable while not wired see next comments

Reply:
Ok, this could definitely be improved by not getting the list of cables / devices / etc. at every frame (we don't need to overload our servers, that's not eco friendly!)
But could be optimized later